### PR TITLE
refactor(dedicated.server): create ftpbackupstorage module

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/multi-az` 
| Bug fix?         | no
| New feature?     | Yes
| Breaking change? | no
| Tickets          | MANAGER-12287 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
Backup Storage refactor:
- Create a new module for ftp backup storage
- move ftp backup access delete to bm-server-component
- move ftp backup access add to bm-server-component
- move ftp backup access update to bm-server-component
- move ftp backup activate to bm-server-component
- move ftp backup order to bm-server-component
- move ftp backup delete to bm-server-component
- move ftp backup password request to bm-server-component

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated.bundle.js [hunk 1]
```diff
@@ -24,16 +24,7 @@ import './dedicated-server/servers/dns/add/dedicated-server-dns-add.controller';
...
-import './dedicated-server/servers/ftp-backup/access/delete/dedicated-server-ftp-backup-access-delete.controller';
-import './dedicated-server/servers/ftp-backup/access/update/dedicated-server-ftp-backup-access-update.controller';
-import './dedicated-server/servers/ftp-backup/activate/dedicated-server-ftp-backup-activate.controller';
-import './dedicated-server/servers/ftp-backup/dedicated-server-ftp-backup.constants';
-import './dedicated-server/servers/ftp-backup/dedicated-server-ftp-backup.controller';
-import './dedicated-server/servers/ftp-backup/dedicated-server-ftp-backup.routes';
-import './dedicated-server/servers/ftp-backup/delete/dedicated-server-ftp-backup-delete.controller';
-import './dedicated-server/servers/ftp-backup/order/dedicated-server-ftp-backup-order.controller';
-import './dedicated-server/servers/ftp-backup/password/request/dedicated-server-ftp-backup-password-request.controller';
+
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated.bundle.js]:
```suggestion
```

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/ftp-backup-storage.controller.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            'server_configuration_ftpbackup_lost_password_failure',
+          ),
+          data,
+          FTP_BACKUP_STORAGE_ALERT,
+        );
+      });
+  }
+
+  goToAdd() {
+    this.$state.go('app.dedicated-server.server.ftpBackup.add');
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/ftp-backup-storage.controller.js:308]:
```suggestion
    return this.$state.go('app.dedicated-server.server.ftpBackup.add');
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-add.routing.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      modal: {
+        component: 'dedicatedServerFtpBackupAccessAdd',
+      },
+    },
+    params: {
+      results: [],
+    },
+    layout: 'modal',
+    resolve: {
+      goBack: /* @ngInject */ (goToFtpBackUpStorage) => goToFtpBackUpStorage,
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-add.routing.js:14]: Shouldn't reload and banner management be added here in the event of success or error?

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-delete.routing.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      modal: {
+        component: 'dedicatedServerFtpBackupAccessDelete',
+      },
+    },
+    params: {
+      ipbackup: null,
+    },
+    layout: 'modal',
+    resolve: {
+      goBack: /* @ngInject */ (goToFtpBackUpStorage) => goToFtpBackUpStorage,
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-delete.routing.js:14]: Shouldn't reload and banner management be added here in the event of success or error?

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-update.routing.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      modal: {
+        component: 'dedicatedServerFtpBackupAccessUpdate',
+      },
+    },
+    params: {
+      ipbackup: null,
+    },
+    layout: 'modal',
+    resolve: {
+      goBack: /* @ngInject */ (goToFtpBackUpStorage) => goToFtpBackUpStorage,
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-update.routing.js:14]: Shouldn't reload and banner management be added here in the event of success or error?

Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-update.routing.js:14]: You have the same thing 3 times, maybe you can directly reuse the goToFtpBackUpStorage that you bind in the component?

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/access/add/dedicated-server-ftp-backup-access-add.controller.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        })
+        .catch(({ data }) => {
+          this.goBack();
+          this.Alerter.alertFromSWS(
+            this.$translate.instant(
+              'server_configuration_ftpbackup_access_add_ip_failure',
+            ),
+            data,
+            FTP_BACKUP_STORAGE_ALERT,
+          );
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/access/add/dedicated-server-ftp-backup-access-add.controller.js:43]: I don't think anything after the goBack has been played. The alert management should be done in transition.

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/activate/dedicated-server-ftp-backup-activate.controller.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  constructor($stateParams, $translate, Server, Alerter) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.Server = Server;
+    this.Alerter = Alerter;
+  }
+
+  $ngOnInit() {
+    this.loading = false;
+  }
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/activate/dedicated-server-ftp-backup-activate.controller.js:13]:
```suggestion
  $onInit() {
    this.loading = false;
  }
```

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/ftp-backup-storage.html [hunk 8]
```diff
@@ -38,29 +38,29 @@
...
+                                    aria-valuenow="{{ $ctrl.ftpBackup.model.usage.value }}"
                                     aria-valuemin="0"
                                     aria-valuemax="100"
                                     data-ng-class="{
-                                        'progress-bar-success': ftpBackup.model.usage === null || ftpBackup.model.usage.value < 51,
-                                        'progress-bar-warning' : ftpBackup.model.usage.value > 50 && ftpBackup.model.usage.value < 81,
-                                        'progress-bar-danger' : ftpBackup.model.usage.value > 80
+                                        'progress-bar-success': $ctrl.ftpBackup.model.usage === null || $ctrl.ftpBackup.model.usage.value < 51,
+                                        'progress-bar-warning' : $ctrl.ftpBackup.model.usage.value > 50 && $ctrl.ftpBackup.model.usage.value < 81,
+                                        'progress-bar-danger' : $ctrl.ftpBackup.model.usage.value > 80
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/ftp-backup-storage.html:47]: Numbers can be set in variables in a constant file

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/delete/dedicated-server-ftp-backup-delete.controller.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+export default class DeleteFtpBackupCtrl {
+  constructor($stateParams, $translate, Server, Alerter) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.Server = Server;
+    this.Alerter = Alerter;
+  }
+
+  $onInit() {
+    this.ftpBackup = this.ipbackup;
```
Comment (darsene) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/delete/dedicated-server-ftp-backup-delete.controller.js:12]: What's the goal here ? If you don't make a copy, it's useless.
Moreover, it hides the use of `ipbackup` in the component

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/delete/dedicated-server-ftp-backup-delete.html [hunk 10]
```diff
@@ -1,19 +1,19 @@
...
+            <div class="text-center" data-ng-if="$ctrl.loading">
                 <oui-spinner></oui-spinner>
             </div>
 
-            <div data-ng-if="!loading">
+            <div data-ng-if="!$ctrl.loading">
                 <p
                     data-translate="server_configuration_ftpbackup_delete_question"
-                    data-translate-values="{ t0: ftpBackup.ftpBackupName }"
+                    data-translate-values="{ t0: $ctrl.ftpBackup.ftpBackupName }"
```
Comment (darsene) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/delete/dedicated-server-ftp-backup-delete.html:16]:
```suggestion
                    data-translate-values="{ t0: $ctrl.ipbackup.ftpBackupName }"
```

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/password-request/dedicated-server-ftp-backup-password-request.controller.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  /* @ngInject */
+  constructor($stateParams, $translate, Server, Alerter) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.Server = Server;
+    this.Alerter = Alerter;
+  }
+
+  $onInit() {
+    this.ftpBackup = this.requestData;
```
Comment (darsene) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/password-request/dedicated-server-ftp-backup-password-request.controller.js:13]: Should be removed

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/password-request/dedicated-server-ftp-backup-password-request.html [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+        data-wizard-title=":: 'server_configuration_ftpbackup_lost_password' | translate"
+    >
+        <div data-wizard-step data-wizard-step-valid="!$ctrl.loading">
+            <div class="text-center" data-ng-if="$ctrl.loading">
+                <oui-spinner></oui-spinner>
+            </div>
+
+            <p
+                data-ng-if="!$ctrl.loading"
+                data-ng-bind="'server_configuration_ftpbackup_lost_password_question' | translate:{ t0: $ctrl.ftpBackup.ftpBackupName }"
```
Comment (darsene) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/password-request/dedicated-server-ftp-backup-password-request.html:15]:
```suggestion
                data-ng-bind="'server_configuration_ftpbackup_lost_password_question' | translate:{ t0: $ctrl.requestData.ftpBackupName }"
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-add.routing.js (hunk 3 above)
Comment (sachinrameshn) [reply packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/ftp-backup/access/dedicated-server-ftp-backup-access-add.routing.js:14]: Seen together no need to take care this.

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/access/add/dedicated-server-ftp-backup-access-add.controller.js [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+          this.goBack();
+        })
+        .finally(() => {
+          this.loading = false;
+        });
+    };
+  }
+
+  addFtpBackup() {
+    return () => {
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/access/add/dedicated-server-ftp-backup-access-add.controller.js:52]: Why you return a function in function ?

File: packages/manager/modules/bm-server-components/src/ftp-backup-storage/ftp-backup-storage.controller.js [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+          ),
+          data,
+          FTP_BACKUP_STORAGE_ALERT,
+        );
+      });
+  }
+
+  goToAdd() {
+    return this.$state.go('app.dedicated-server.server.ftpBackup.add');
+  }
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/ftp-backup-storage/ftp-backup-storage.controller.js:313]: Maybe you can move in routing file ?

**Review by ghyenne — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**